### PR TITLE
Update outdated java example in wiki #6331

### PR DESCRIPTION
**Pull request by ozcanyus**
Updated java example in docs/How-To-Use-RxJava.md file with java 8 version.

Resolves: #6331

### CODE REVIEW

File: docs/How-To-Use-RxJava.md [hunk 1]
```diff
@@ -12,14 +12,7 @@ You can find additional code examples in the `/src/examples` folders of each [la
...
 public static void hello(String... names) {
-    Observable.from(names).subscribe(new Action1<String>() {
-
-        @Override
-        public void call(String s) {
-            System.out.println("Hello " + s + "!");
-        }
-
-    });
+    Observable.fromArray(names).subscribe(s -> System.out.println("Hello " + s));
```
Comment (akarnokd) [inline docs/How-To-Use-RxJava.md]: I'd instead show a `Consumer<String>` and the lambda version, similar to the [hello world](https://github.com/ReactiveX/RxJava#hello-world) example.

Comment (ozcanyus) [reply docs/How-To-Use-RxJava.md]: Updated similar to hello world example.

File: docs/How-To-Use-RxJava.md [hunk 2]
```diff
@@ -11,22 +11,21 @@ You can find additional code examples in the `/src/examples` folders of each [la
...
-public static void hello(String... names) {
-    Observable.from(names).subscribe(new Action1<String>() {
-
-        @Override
-        public void call(String s) {
-            System.out.println("Hello " + s + "!");
-        }
-
-    });
+public static void main(String[] args) {
```
Comment (akarnokd) [inline docs/How-To-Use-RxJava.md]: No, I meant show codes as anonym inner class and lambda, but keep the essence of the original example, i.e. fromArray

**Review by akarnokd — CHANGES_REQUESTED** [review]

File: docs/How-To-Use-RxJava.md (hunk 2 above)
Comment (ozcanyus) [reply docs/How-To-Use-RxJava.md]: Oh ok, i'll fix.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]

**Pull request merged**